File: JavaSource/src/main/java/ru/akhcheck/patterns/decorator/notifiers/EmailNotifier.java
package ru.akhcheck.patterns.decorator.notifiers;

/** Добавляет отправку письмом поверх вложенного отправителя. */
public class EmailNotifier extends DecoratorNotifier {
    private final String address;

    public EmailNotifier(Notifier notifier, String address) {
        super(notifier);
        this.address = address;
    }

    @Override
    public void notify(String message) {
        super.notify(message);
        System.out.println("[email to=" + address + "] " + message);
    }
}

File: Decorator/python-source/notifiers/EmailNotifier.py
from __future__ import annotations

from .DecoratorNotifier import DecoratorNotifier
from .Notifier import Notifier


class EmailNotifier(DecoratorNotifier):
    """Добавляет отправку письмом поверх вложенного отправителя."""

    def __init__(self, notifier: Notifier, address: str = "team@example.com") -> None:
        super().__init__(notifier)
        self._address = address

    def notify(self, message: str) -> None:
        super().notify(message)
        print(f"[email to={self._address}] {message}")
